rclcpp/src/rclcpp/time_source.cpp

```diff
@@ -167,39 +169,28 @@ void TimeSource::clock_cb(const builtin_interfaces::msg::Time::SharedPtr msg)
 
 void TimeSource::on_parameter_event(const rcl_interfaces::msg::ParameterEvent::SharedPtr event)
 {
-  for (auto & new_parameter : event->new_parameters) {
-    if (new_parameter.value.type == rclcpp::parameter::ParameterType::PARAMETER_BOOL) {
-      if (new_parameter.name == "use_sim_time") {
-        bool value = new_parameter.value.bool_value;
-        if (value) {
-          parameter_state_ = SET_TRUE;
-          enable_ros_time();
-        } else {
-          parameter_state_ = SET_FALSE;
-          disable_ros_time();
-        }
-      }
+  rclcpp::ParameterEventsFilter filter(event, {"use_sim_time"},
```

Comment:
This API is a bit unwieldy, which is not a bad thing (at least I can't think of a better way to do it), it's just not obvious at first glance, maybe a comment here to say something like "filter for new or changed events of use_sim_time" would help to make it more obvious when someone is reading this code.